internal/logging/logging_test.go

```diff
@@ -0,0 +1,129 @@
+package logging
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/apex/log"
+	"github.com/fatih/color"
+	"github.com/sclevine/spec"
+)
+
+const testTime = "2019/05/15 01:01:01.000000"
+
+func newTestLogger(stdout, stderr io.Writer) *logWithWriters {
+	hnd := &handler{
+		writer: stdout,
+		timer: func() time.Time {
+			tm, _ := time.Parse(timeFmt, testTime)
+			return tm
+		},
+	}
+	var lw logWithWriters
+	lw.handler = hnd
+	lw.out = hnd.writer
+	lw.errOut = stderr
+	lw.Logger.Handler = hnd
+	lw.Logger.Level = log.DebugLevel
+	return &lw
+}
+
+func TestPackCLILogger(t *testing.T) {
+	spec.Run(t, "PackCLILogger", func(t *testing.T, when spec.G, it spec.S) {
+		var log, errlog bytes.Buffer
+		var logger *logWithWriters
+
+		it.Before(func() {
+			color.NoColor = false
+			logger = newTestLogger(&log, &errlog)
+		})
+
+		it.After(func() {
+			log.Reset()
+			errlog.Reset()
+		})
+
+		it("can enable time in logs", func() {
+			logger.WantTime(true)
+			logger.Error("test")
+			expected := "2019/05/15 01:01:01.000000 \x1b[31;1mERROR  \x1b[0mtest                     \n"
+			if expected != log.String() {
```

Comment:
We have some custom assertions in `test_helpers.go` that you can use instead. We typically import them under the alias `h`. Look for `h.AssertEqual` for instance.